source/Commands/BirthdayCommand.ts

```diff
@@ -0,0 +1,305 @@
+import { AbstractCommand } from "./AbstractCommand";
+import { AbstractCommandOptions } from "../Material/AbstractCommandOptions";
+import { Client, Message, MessageEmbed, GuildMember, Guild, Role } from "discord.js";
+import { UserService } from "../Service/UserService";
+import { User } from "../Material/User";
+import { PermissionLevel } from "../Material/PermissionLevel";
+import { stringify } from "querystring";
+import { Brackets, MongoClient } from "typeorm";
+import { ServerData } from "../Material/ServerData";
+import { ServerDataService } from "../Service/ServerDataService";
+import { HelperFunctions } from "../Material/HelperFunctions";
+
+export class BirthdayCommand extends AbstractCommand {
+	public commandOptions: BirthdayCommandHelp = new BirthdayCommandHelp();
+	public userService: UserService = UserService.getInstance();
+	public serverDataService: ServerDataService = ServerDataService.getInstance();
+
+	// Verifies that this is a valid date
+	public static validateBirthday(date: number, month: number): boolean {
+		let daysInMonths = [0, // So the array can be 1-indexed
+			31, 29, 31, 30,
+			31, 30, 31, 31,
+			30, 31, 30, 31
+		];
+		if (month > 0 && month < 13 && date > 0 && date <= daysInMonths[month]) {
+			return true
+		} else {
+			return false;
+		}
+	}
+
+	public async runInternal(bot: Client, message: Message, messageArray: string[]) {
+		let user: User = await this.userService.getUser(message.member);
+		let server: ServerData = await this.serverDataService.getServerData(message.guild);
+		let id: string = message.author.id;
+
+		let numberSuffix = ["th", "st", "nd", "rd", "th", "th", "th", "th", "th", "th"]; // 1st, 2nd, 3rd etc.
+		let monthNames = ["December", "January", "Febuary", "March", "April", "May", "June", "July", "August", "September", "October", "November"];
+		if (messageArray[0]) {
+			switch (messageArray[0]) {
+				case "set":
+					if (messageArray.length == 4) { // Setting a birthday by ID/tag
+						let taggedID: string = HelperFunctions.getID(messageArray[1]);
+						let month = parseInt(messageArray[3]);
+						let date = parseInt(messageArray[2]);
+						if (taggedID != user.discordID) { // Setting somebody else's birthday
+							if (user.permissionLevel >= PermissionLevel.moderator) {
+								if (HelperFunctions.verifyUserID(message.guild, taggedID)) { // A staff member setting somebody else's birthday
+									if (BirthdayCommand.validateBirthday(date, month)) {
+										let taggedUser: User = await this.userService.getUserWithID(taggedID);
+										if (!taggedUser.birthday) taggedUser.birthday = new Date();
+										taggedUser.birthday.setMonth(month);
+										taggedUser.birthday.setDate(date);
+										taggedUser.save();
+
+										message.channel.send(`<@!${taggedID}>'s birthday has been set.`);
+									} else {
+										message.channel.send("Please provide a valid date.");
```

Comment:
When an action fails like in this case please send an embed.
To get the default failed embed you can use `super.getFailedEmbed()` and set the description